### PR TITLE
GUI

### PR DESCRIPTION
**Pull request by geralddejong**

### CODE REVIEW

File: src/build/growth.rs [hunk 1]
```diff
@@ -65,7 +65,7 @@ impl Growth {
 
     pub fn init(&mut self, fabric: &mut Fabric) {
         let BuildPhase { seed, root } = &self.plan.build_phase;
-        let node = root.clone().unwrap();
+        let node = root.clone().unwrap_or(Grow {forward: "".into(), scale_factor: 1.0, post_growth_node: None});
```
Comment (mitchelljustin) [inline src/build/growth.rs:68]: Aw really? I'd at least move this to a impl of Default

File: src/build/tenscript/mod.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
-use std::collections::HashMap;
 use std::fmt::{Display, Formatter};
+use hashbrown::HashMap;
```
Comment (mitchelljustin) [inline src/build/tenscript/mod.rs:2]: So goofy that this is necessary.

File: src/controls/fabric_choice.rs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+use iced::{Alignment, Length};
+use iced_wgpu::Renderer;
+use iced_winit::Color;
+use iced_winit::widget::{Button, Row, Text};
+use crate::build::tenscript::fabric_plan;
+use crate::controls::{Action, Message};
+
+#[derive(Clone, Debug)]
+pub enum FabricChoiceMessage {
+    ChooseFabric(&'static str),
```
Comment (mitchelljustin) [inline src/controls/fabric_choice.rs:11]: Hmm, I wonder if you want to make these Strings instead to select runtime created fabric plans.

File: src/controls/fabric_choice.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                            None => {
+                                Color::WHITE
+                            }
+                            Some(current) => {
+                                if choice == current {
+                                    Color::WHITE
+                                } else {
+                                    Color::from_rgb(0.0, 1.0, 0.0)
+                                }
+                            }
```
Comment (mitchelljustin) [inline src/controls/fabric_choice.rs:56]: Remove braces in match arms

File: src/controls/mod.rs [hunk 5]
```diff
@@ -177,54 +184,67 @@ impl GUI {
...
 
-impl Controls {
+impl ControlState {
     pub fn take_actions(&self) -> Vec<Action> {
         self.action_queue.borrow_mut().split_off(0)
     }
 
-    pub fn strain_threshold(&self, maximum_strain: f32) -> f32 {
-        maximum_strain * self.strain_nuance
+    pub fn get_strain_threshold(&self, maximum_strain: f32) -> f32 {
```
Comment (mitchelljustin) [inline src/controls/mod.rs:230]: I'm partial to 
```suggestion
    pub fn strain_threshold(&self, maximum_strain: f32) -> f32 {
```

File: src/controls/mod.rs [hunk 6]
```diff
@@ -277,31 +298,13 @@ impl Program for Controls {
...
-                                    Text::new(format!("{strain_limit:.05}"))
-                                        .style(Color::WHITE)
-                                )
-                                .push(
-                                    Button::new(Text::new("Add Pulls"))
-                                        .on_press(Message::AddPulls)
-                                )
-                        }
+                        Showing::FabricChoice => self.fabric_choice_control.view(),
+                        Showing::StrainThreshold => self.strain_threshold_control.view(),
```
Comment (mitchelljustin) [inline src/controls/mod.rs:302]: Noice

**Review by mitchelljustin — APPROVED** [review]

File: src/build/growth.rs (hunk 1 above)
Comment (geralddejong) [reply src/build/growth.rs:68]: This stuff is in transition, sorry eh.

File: src/controls/fabric_choice.rs (hunk 3 above)
Comment (geralddejong) [reply src/controls/fabric_choice.rs:11]: Seems to add a lot of noise where it's used

File: src/build/tenscript/mod.rs (hunk 2 above)
Comment (geralddejong) [reply src/build/tenscript/mod.rs:2]: Goofy is not the word!  We need to get the word out.

**Pull request merged**